agent/api/registryauth.go

```diff
@@ -36,6 +37,11 @@ type ECRAuthData struct {
 	lock             sync.RWMutex
 }
 
+type ASMAuthData struct {
```

Comment:
lint: you're missing comments for a number of exported types and fields throughout. Please fix all of them